sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2SQLSuite.scala

```diff
@@ -482,6 +482,8 @@ class DataSourceV2SQLSuite extends QueryTest with SharedSparkSession with Before
 
     val rdd = sparkContext.parallelize(table.asInstanceOf[InMemoryTable].rows)
     checkAnswer(spark.internalCreateDataFrame(rdd, table.schema), spark.table("source"))
+
+    spark.sql("DROP TABLE table_name")
```

Comment:
why is this need? This test suite always clear all the tables after each test case.

Reply:
I agree

Reply:
These tests create a `table_name` table with v1 catalog (db = default, isTemporary = false). Is there a recommended way to clean this up?

Reply:
In `after`, the catalog manager is reset:

```
spark.sessionState.catalogManager.reset()
```

That discards all of the in-memory catalogs and the tables in them so test cases won't conflict.

Reply:
Ah, my mistake. This can be removed, but the `DROP TABLE` below is still valid.

For the test below, when the table `table_name` is created, V1 catalog (`SessionCatalog`) is used via `sparkSession.sessionState.catalog` So, it cannot be reset from `spark.sessionState.catalogManager.reset()`.

I think `spark.sessionState.catalog.reset()` is the right way to do in `after`, right?